src/main/java/sinatra/Sinatra.java

```diff
@@ -0,0 +1,147 @@
+package sinatra;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the main Sinatra application.
+ */
+public class Sinatra {
+
+    private ArrayList<Task> tasks;
+    private Storage storage;
+    private static final String FILE_PATH = "tasks.txt";
+
+    /**
+     * Constructs a new Sinatra object, initializes tasks, prints the introduction,
+     * loads tasks from storage, and starts the scanner for user input.
+     */
+    public Sinatra() {
+        this.tasks = new ArrayList<Task>();
+        this.printIntro();
+        this.storage = new Storage(Sinatra.FILE_PATH);
+        this.tasks = storage.loadTasksFromFile();
+        this.sinatraScanner();
+    }
+
+    /**
+     * Prints the introduction message.
+     */
+    private void printIntro() {
+        String initial = "Hello! I'm Sinatra. \nWhat can I do for you? ";
+        System.out.println(initial);
+    }
+    private ArrayList<Task> findTasksWithContent(String contentPart) {
+
+        ArrayList<Task> foundTasks = new ArrayList<Task>();
+        for (Task task : tasks) {
+            if (task.getContent().contains(contentPart)) {
+                foundTasks.add(task);
+            }
+        }
+        return foundTasks;
+    }
+
+    /**
+     * Starts the scanner to handle user inputs.
+     */
+    private void sinatraScanner() {
+        Scanner scanner = new Scanner(System.in);
+        while (true) {
+            String input = scanner.nextLine();
+            handleInputs(input);
+        }
+    }
+
+    /**
+     * Handles the user inputs and performs the corresponding actions.
+     *
+     * @param message the user input message
+     */
+    private void handleInputs(String message) {
```

Comment:
Have you considered making a separate class for this (e.g. Parser)? Doing so could encapsulate OOP better